[CS2103-T14-1] Teacher's Notebook

weikiat97 opened the pull request:
@Xelyion
@SebastianLie
@sheryltay28
@weikiat97

jyx11011 commented on docs/AboutUs.adoc:
```diff
@@ -4,51 +4,51 @@
+{empty}[https://github.com/sheryltay28[github]] [<<johndoe#, portfolio>>]
 
 Role: Developer +
 Responsibilities: Dev Ops + Threading
```
Could modify the responsibility of each member.

ZhangHuafan commented on docs/UserGuide.adoc:
```diff
@@ -45,133 +35,451 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-=== Viewing help : `help`
+// tag::add[]
+== Adding: add
+
```
Maybe consider using "=== Addding: add" since these are the subsections under "Feature"

ZhangHuafan commented on docs/UserGuide.adoc:
```diff
@@ -45,133 +35,451 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+=== Classes
+
+Adds a class to the class list. +
+Format: add c/CLASS
```
Maybe consider using `` for command keyword so that it is easier to spot a command line to use.

whneo97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -349,6 +398,10 @@ _{More to be added}_
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should be able to display student's photos in 4KKKKKK resolutionnnnn
+.  Application should respond within 5 seconds
+.  Should have a clean and minimalist interface
```
Clean and minimalist interface may not be well defined (it's hard to determine when an interface is clean or minimalist enough)

whneo97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -349,6 +398,10 @@ _{More to be added}_
+.  Should be able to display student's photos in 4KKKKKK resolutionnnnn
+.  Application should respond within 5 seconds
+.  Should have a clean and minimalist interface
+.  User should be able to view and undo changes
```
This should perhaps be a functional requirement?

jyx11011 commented on docs/UserGuide.adoc:
```diff
@@ -45,133 +35,451 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-e.g. `find James Jake`
-* *List* : `list`
-* *Help* : `help`
+Teacher’s Notebook is for teachers who prefer to make use of an application to keep track of the deliverables, tasks and students’ information. It is optimized for teachers who prefer to work with a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you are a teacher who wants to manage your students and your tasks efficiently, and can type fast,  ...
```
It seems like a duplicate from this line onwards

whneo97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +365,31 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+=== Use case: Delete person
```
Perhaps you have forgotten to push the updated commit? It seems like this part is a duplicate of the original template given.

ZhangHuafan commented on docs/AboutUs.adoc:
```diff
@@ -4,51 +4,51 @@
+{empty}[https://github.com/weikiat97[github]] [<<johndoe#, portfolio>>]
 
 Role: Team Lead +
 Responsibilities: UI
```
Maybe you can change the roles to correct ones =)

ZhangHuafan reviewed (commented):
Hi! It looks that your project is interesting and useful. Look forward to seeing your final product. However, you may want to remove the extra code in the user guide to make it neat! Besides, you can close the issues which you have done. All the best!

SebastianLie committed:
Merge pull request #167 from SebastianLie/master

Fixed issue #100 in add and deleteclass

sheryltay28 committed:
not working code

sheryltay28 committed:
added minor changes to make it finally work

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into lesson-json

# Conflicts:
#	src/main/java/seedu/address/model/Notebook.java
#	src/main/java/seedu/address/model/lesson/Time.java

sheryltay28 committed:
checkstyle issues

weikiat97 committed:
Fixed issue #113 (added check for whether phone and parent phone are the same).

sheryltay28 committed:
fixed minor changes in editLesson command

sheryltay28 committed:
removed debugging line

sheryltay28 committed:
Merge pull request #168 from sheryltay28/lesson-json

Lesson json

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into fixing-issues

sheryltay28 committed:
changed edit lesson message usage

sheryltay28 committed:
edited delete lesson command

sheryltay28 committed:
edited user guide to reflect changes in delete lesson command

sheryltay28 committed:
fixed minor writing to json issue

Commit:
Change upload functionality

weikiat97 committed:
Fixed issue #110

weikiat97 committed:
Update user guide with undo / redo.

weikiat97 committed:
Added ppp (not completed).

SebastianLie committed:
Merge pull request #169 from SebastianLie/master

Change upload functionality

weikiat97 committed:
Made another change to the tagset.

weikiat97 committed:
Merge pull request #170 from weikiat97/fix-trivial-issues-and-ppp

Fix trivial issues and updated ppp.

sheryltay28 committed:
checkstyle issues

weikiat97 committed:
Merge pull request #171 from sheryltay28/fixing-issues

Fixing issues

weikiat97 committed:
Populated SampleDataUtil with more classes, lessons and students.

weikiat97 committed:
Fixed travis and fixed the delete lesson command instructions in command master list.

weikiat97 committed:
Merge pull request #172 from weikiat97/master2

Populated SampleDataUtil with more classes, lessons and students.

weikiat97 committed:
Refactored out GetUnsubmittedAssignmentsCommand from GetStudentGradesCommand.

weikiat97 committed:
Updated UI picture with populated data.
Fixed the GUI such that it scales properly now.

weikiat97 committed:
Fixed bug where a corrupted notebook.json cannot be opened.

weikiat97 committed:
Assignment deadlines must not be before the current system time.

weikiat97 committed:
Changed description of isValidGrade

weikiat97 committed:
Added lesson comparator and some proposed features.

weikiat97 committed:
Merge pull request #174 from weikiat97/master2

Fixed bug for corrupted notebook.json and allow GUI to scale properly.

weikiat97 committed:
Merge pull request #173 from weikiat97/refactored-out-getunsubmitted

Refactored GetUnsubmittedAssignmentsCommand from GetStudentGradesCommand.

weikiat97 committed:
Merge pull request #175 from weikiat97/restricted-deadline-for-assignments

Restricted deadline for assignments

sheryltay28 committed:
fixed minor json error for lesson

weikiat97 committed:
Updated README with Coveralls and Appveyor, and set up the applications as well.

Xelyion committed:
Added color for current classroom

weikiat97 committed:
Merge pull request #176 from weikiat97/master2

Added Coveralls and Appveyor.

Xelyion committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

# Conflicts:
#	src/main/resources/view/MainWindow.fxml

Xelyion committed:
Merge pull request #177 from Xelyion/master

Added color to current classroom

weikiat97 committed:
Updated README.adoc

weikiat97 committed:
Edited clear command. Updated UG.

weikiat97 committed:
Merge pull request #179 from weikiat97/master2

Edited clear command. Updated UG.

sheryltay28 committed:
fixed reminder issues

sheryltay28 committed:
cleaned code

sheryltay28 committed:
fixed checkstyle issues

sheryltay28 committed:
changed header to "lessons"

sheryltay28 committed:
Merge branch 'master' into fixing-reminder

sheryltay28 committed:
Merge pull request #180 from sheryltay28/fixing-reminder

Fixed reminder issues

weikiat97 committed:
Changed Main Window scalability

sheryltay28 committed:
edit student card

Commit:
Add reset picture command

Commit:
Add leftover javadoc

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

SebastianLie committed:
Merge pull request #181 from SebastianLie/master

Add command to reset display picture

weikiat97 committed:
Changed the setting for lessons to fix undo redo bug, and made duplicate student more strict.

sheryltay28 committed:
fixed exit issues

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into fix-exit

# Conflicts:
#	src/main/java/seedu/address/ui/StudentCard.java

sheryltay28 committed:
Merge pull request #183 from sheryltay28/fix-exit

Fix exit

Commit:
Add remove listener functions

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

# Conflicts:
#	src/main/java/seedu/address/ui/MainWindow.java

Commit:
Add PPP

weikiat97 committed:
Changed window size

weikiat97 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into master2

weikiat97 committed:
Updated UG for Students.

weikiat97 committed:
Fixed travis test.

weikiat97 committed:
Merge pull request #182 from weikiat97/master2

Fix undo redo bug and made isSameStudent defined as same name.

weikiat97 committed:
Halfway through editing DG's current PUML diagrams to fit our current system.

weikiat97 committed:
Edited DG to suit current system.

weikiat97 committed:
Merge pull request #184 from weikiat97/change-ug-8th-nov

Added student part for ug.

weikiat97 committed:
Merge pull request #185 from weikiat97/change-dg-8th-nov

Change DG to suit current system.

sheryltay28 committed:
show day of week when displaying lesson

weikiat97 committed:
Added tests for students and AddStudentCommandTest.

sheryltay28 committed:
added find command for lesson

Commit:
Add draft of PPP

weikiat97 committed:
Fixed preexisting tests for logic package, except for 6 tests (will do tomorrow).

sheryltay28 committed:
removed Reminder class and moved Scheduler to UI

sheryltay28 committed:
updated command masterlist

Commit:
Touch up autocomplete and User Guide

Commit:
fix travis errors

Commit:
Fixed asciidoctor issues

Commit:
Fix no newline at EOF

Commit:
Fix uncaught exception in autocomplete

Commit:
Cleaned up code

Commit:
Fix checkstyle errors

Commit:
Cleaned up code further

Commit:
Added more sensible assertions

SebastianLie committed:
Merge pull request #187 from SebastianLie/master

Add PPP, touch up autocomplete

sheryltay28 committed:
prevent clash of lesson timings

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into adding-find-lesson

sheryltay28 committed:
fixed checkstyle issues

sheryltay28 committed:
Merge pull request #188 from sheryltay28/adding-find-lesson

Adding find lesson and fixing reminder issues

Commit:
Add initial display picture test

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

Commit:
Touch up autocomplete and add tests

SebastianLie committed:
Merge pull request #189 from SebastianLie/master

Touch up autocomplete and add tests

sheryltay28 committed:
created lesson test

Commit:
Add more tests

Commit:
Added Javadoc comment

Commit:
Add more tests

Commit:
Uncommented some tests

Commit:
Add List Student tests

SebastianLie committed:
Merge pull request #190 from SebastianLie/master

Add more tests

weikiat97 committed:
2 Failing Cases from DeleteStudentCommandTest and 1 from JsonNotebookStorageTest.

weikiat97 committed:
Merge branch 'master' into add-tests-for-student

weikiat97 committed:
Fix travis.

weikiat97 committed:
Fixed travis.

weikiat97 committed:
Fixed Travis.

weikiat97 committed:
Fixed merging issues.

weikiat97 committed:
Fixed Travis Errors.
JsonNotebookStorageTest 72
JsonNotebookStorageTest 79
JsonNotebookStorageTest 85
DeleteStudentCommandTest 40
DeleteStudentCommandTest 66
Yet to pass.

weikiat97 committed:
Merge pull request #186 from weikiat97/add-tests-for-student

Added some tests for students, logic and edited preexisting ones.

Commit:
Add changes to PPP

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

Commit:
Add tests in logic, storage, model

Commit:
Add tests for unique assignment list

Commit:
Add tests for unique assignment list; ignore previous commit

Commit:
Corrected Checkstyle errors

Commit:
Scaled images in user guide

SebastianLie committed:
Merge pull request #191 from SebastianLie/master

Add tests in logic, storage, model

weikiat97 committed:
Finished model.classroom tests

Commit:
Add Upload Command  implementation to developer guide

weikiat97 committed:
Added some tests for logic.command for assignments, model.classroom, Caretaker, Memento and Notebook

weikiat97 committed:
Merge branch 'master' into add-tests-for-student

weikiat97 committed:
Update ResetPicCommandTest.java

weikiat97 committed:
Update ResetPicCommandParserTest.java

weikiat97 committed:
Update ResetPicCommandTest.java

weikiat97 committed:
Update UniqueAssignmentListTest.java

weikiat97 committed:
Update UniqueAssignmentListTest.java

weikiat97 committed:
Update TypicalAssignments.java

weikiat97 committed:
Update TypicalIndexes.java

weikiat97 committed:
Merge pull request #192 from weikiat97/add-tests-for-student

Add tests for model.classroom, logic.commands, Caretaker, Memento and Notebook

weikiat97 committed:
Fixed merge conflicts when refactoring ASSIGNMENT_ONE, ASSIGNMENT_TWO, INDEX_FIRST_STUDENT, INDEX_SECOND_STUDENT, INDEX_THIRD_STUDENT.

weikiat97 committed:
Merge pull request #193 from weikiat97/master2

Fixed merge conflicts.

weikiat97 committed:
Fixed undo-redo bug.

weikiat97 committed:
Merge pull request #194 from weikiat97/update-dg-last-day

Fixed undo-redo bug.

sheryltay28 committed:
added class name test

Commit:
Finish 1st draft of PPP

Commit:
Add minor change to PPP

sheryltay28 committed:
added ppp

Commit:
Add tests for unique lesson list

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

Commit:
Fixed checkstyle errors

Commit:
Fixed EOF checkstyle error

Xelyion committed:
Removed header from template

SebastianLie committed:
Merge pull request #195 from SebastianLie/master

Add 1st real draft of PPP

Xelyion committed:
Put the lines in header back

They do not aid in removing header from website. Other groups did it without modifying this file.

Xelyion committed:
Merge pull request #10 from AY1920S1-CS2103-T14-1/master

Merge with team repo

weikiat97 committed:
Updated DG, UG and PPP.

weikiat97 committed:
Merge pull request #196 from weikiat97/adding-dg-parts

Updated DG, UG and PPP.

sheryltay28 committed:
added to user guide and edited command masterlist

Commit:
Added full upload command tests

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into adding-find-lesson

# Conflicts:
#	src/test/java/seedu/address/logic/commands/CommandTestUtil.java
#	src/test/java/seedu/address/testutil/LessonBuilder.java
#	src/test/java/seedu/address/testutil/TypicalLessons.java

sheryltay28 committed:
edited ppp

Xelyion committed:
Fixed bug in assignments

Xelyion committed:
Merge pull request #11 from AY1920S1-CS2103-T14-1/master

Merge with team repo

Xelyion committed:
Fixed bugs in Assignment and added PPP.

Xelyion committed:
Removed comment

sheryltay28 committed:
fixed minor errors

sheryltay28 committed:
Merge pull request #197 from sheryltay28/adding-find-lesson

added ppp, edited user guide and added tests

Commit:
Touch up autocomplete and UserGuide

Commit:
Fix checkstyle errors

SebastianLie committed:
Merge pull request #200 from SebastianLie/master

Touch up autocomplete

sheryltay28 committed:
removed listlessons

Xelyion committed:
Updated UG and added "Late to the party"

Xelyion committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

weikiat97 committed:
Made documentation changes and javadocs comments. Made the overall code neater and more readable to future developers.

sheryltay28 committed:
editing user guide

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into delete-list

weikiat97 committed:
Merge branch 'master' into last-day-

weikiat97 committed:
Fixed Travis.

weikiat97 committed:
Merge pull request #202 from weikiat97/last-day-

Made documentation changes and overall code base neater.

sheryltay28 committed:
Merge pull request #203 from sheryltay28/delete-list

Delete list

Commit:
Added tests for SetClassroomParser

sheryltay28 committed:
added SchedulingActivityDiagram

sheryltay28 committed:
dg puml

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into delete-list

sheryltay28 committed:
checkstyle issues

sheryltay28 committed:
Merge pull request #204 from sheryltay28/delete-list

edited dg

sheryltay28 committed:
added to dg

Commit:
Add more logic tests

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

# Conflicts:
#	src/main/java/seedu/address/logic/parser/SetClassroomCommandParser.java

Commit:
Fix checkstyle errors

SebastianLie committed:
Merge pull request #205 from SebastianLie/master

Add more logic tests

weikiat97 committed:
Added JsonSerializableClassroomTest.

Xelyion committed:
Added equals method to Assignment

weikiat97 committed:
Added JsonAdaptedAssignmentTest.

Xelyion committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

# Conflicts:
#	src/main/java/seedu/address/model/assignment/Assignment.java
#	src/main/java/seedu/address/model/assignment/AssignmentGrades.java
#	src/main/java/seedu/address/model/classroom/Classroom.java

sheryltay28 committed:
finished adding lesson tests

Xelyion committed:
Fixed ascii errors

Xelyion committed:
Merge pull request #201 from Xelyion/master

Added PPP and updated UG

weikiat97 committed:
Fixed issue where the assignments were unable to update the new students name after editing.

weikiat97 committed:
Removed the equality for Assignment for merge.

weikiat97 committed:
Removed unnecessary methods.

weikiat97 committed:
Fixed travis.

weikiat97 committed:
Merge branch 'Fix-edit-student' of https://github.com/weikiat97/main into Fix-edit-student

weikiat97 committed:
Fixed travis.

weikiat97 committed:
Fixed more travis.

weikiat97 committed:
Fixed more travis.

weikiat97 committed:
Please fix travis.

weikiat97 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into Fix-edit-student

Xelyion committed:
Updated PPP

weikiat97 committed:
Merge branch '1213' of https://github.com/weikiat97/main into Fix-edit-student

weikiat97 committed:
Fixed travis.

weikiat97 committed:
Merge pull request #207 from weikiat97/Fix-edit-student

Fixed issue of editing students.

weikiat97 committed:
Added UndoCommandTest and RedoCommandTest.

weikiat97 committed:
Merge pull request #208 from weikiat97/last-push

Added UndoCommandTest and RedoCommandTest.

Commit:
Fix minor autocomplete bugs

sheryltay28 committed:
added addLessonCommand tests

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into add-tests

weikiat97 committed:
Updated UserGuide with all the commands that were left out, and added Command Summary.

Xelyion committed:
Added to DG

sheryltay28 committed:
checkstyle issues

sheryltay28 committed:
Merge pull request #210 from sheryltay28/add-tests

Add tests

sheryltay28 committed:
updated user guide and ppp

sheryltay28 committed:
edited ppp

weikiat97 committed:
Merge pull request #209 from weikiat97/last-push

Update UG.

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into more-tests

Xelyion committed:
Improved DG

Xelyion committed:
Update DeveloperGuide.adoc

Xelyion committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

weikiat97 committed:
Updated DG.

weikiat97 committed:
Merge pull request #211 from weikiat97/last-push

Updated DG.

Xelyion committed:
Merge pull request #212 from Xelyion/master

Updated DG (Assignment)

sheryltay28 committed:
fixed reminder

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into more-tests

Commit:
Touch up PPP, add command test

sheryltay28 committed:
added deleteLesson

Xelyion committed:
Edited About Us to reflect PPP

Xelyion committed:
Merge pull request #213 from Xelyion/master

Edited About Us to reflect PPP

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

# Conflicts:
#	docs/UserGuide.adoc

sheryltay28 committed:
last min stuff

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into more-tests

Commit:
Fix checkstyle

Xelyion committed:
Update Xelyion.adoc

sheryltay28 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main into more-tests

Commit:
Removed errors in user guide

sheryltay28 committed:
fixed checkstyle

SebastianLie committed:
Merge pull request #214 from SebastianLie/master

Add final changes to autocomplete

sheryltay28 committed:
Merge pull request #215 from sheryltay28/more-tests

last min stuff

weikiat97 committed:
Updated Ui.png

weikiat97 committed:
Merge pull request #216 from weikiat97/last-push

Updated Ui.png

Xelyion committed:
Update AboutUs.adoc

Commit:
Updated Docs

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-T14-1/main

Commit:
Finish up PPP

sheryltay28 committed:
added png files

Xelyion committed:
Update Xelyion.adoc

sheryltay28 committed:
Merge pull request #218 from sheryltay28/png-files

added png files

SebastianLie committed:
Merge pull request #217 from SebastianLie/master

Add minor PPP changes
